fitbenchmarking/controllers/aoclda_controller.py

```diff
@@ -0,0 +1,327 @@
+        elif self.minimizer == "newton":
+            model = b"quasi-newton"  # self._options[b"model"] = 2
+            method = b"galahad"  # self._options[b"nlls_method"] = 4
+            glob_strategy = b"tr"  # self._options[b"type_of_method"] = 1
+        elif self.minimizer == "newton_reg":
+            model = b"quasi-newton"  # self._options[b"model"] = 2
+            method = b"galahad"  # self._options[b"nlls_method"] = 4
+            glob_strategy = b"reg"  # self._options[b"type_of_method"] = 2
+        elif self.minimizer == "newton-tensor":
+            model = b"tensor-newton"  # self._options[b"model"] = 4
+            method = b"galahad"  # self._options[b"nlls_method"] = 4
+            glob_strategy = b"tr"  # self._options[b"type_of_method"] = 1
+        elif self.minimizer == "newton-tensor_reg":
+            model = b"tensor-newton"  # self._options[b"model"] = 4
+            method = b"galahad"  # self._options[b"nlls_method"] = 4
+            glob_strategy = b"reg"  # self._options[b"type_of_method"] = 2
+        else:
+            raise UnknownMinimizerError(
+                f"No {self.minimizer} minimizer for AOCLDA"
+            )
+
+        # If parameter ranges have been set in problem, then set up bounds
+        # option. For AOCLDA, this must be a 2 tuple array like object,
+        # the first tuple containing the lower bounds for each parameter
+        # and the second containing all upper bounds.
+        if self.value_ranges is not None:
+            self._lb, self._ub = zip(*self.value_ranges)
+
+        self._handle = nlls(
+            n_coef=n_coef,
+            n_res=n_res,
+            weights=None,
+            lower_bounds=self._lb,
+            upper_bounds=self._ub,
+            order="c",
+            model=model,
+            method=method,
+            glob_strategy=glob_strategy,
+            verbose=self.VERBOSE,
+        )
+
+        # Set initial iterate
+        self._x = np.array(
+            self.initial_params
+        )  # x has to be np.array and is overwritten!
+
+    def fit(self):
+        """
+        Run problem with AOCLDA.
+        Note: only this method is timed
+        """
+
+        try:
+            self._handle.fit(
+                x=self._x,
+                fun=self.eval_r,
+                jac=self.aoclda_jac,
+                hes=self.aoclda_hes,
+                hep=None,
+                data=self,
+                maxit=self.maxit,
+                xtol=self.xtol,
+                # ftol=self.ftol,
+                # abs_ftol=self.abs_ftol,
+                # gtol=self.gtol,
+                # abs_gtol=self.abs_gtol,
+                # fd_step = self.fd_step,
+                # fd_ttol = self.fd_ttol
+            )
+
+        except RuntimeWarning:  # FIXME this is not caught
+            #    2: Software run but didn't converge to solution
+            #    6: Solver has exceeded maximum allowed runtime
+            self._status = 2
```

Comment:
For the max runtime setting, we shouldn't have to raise exceptions within each individual controller, as this is handled [here](https://github.com/fitbenchmarking/fitbenchmarking/blob/f9b05d41dfe5c104288528ed64ce61080e8da7ce/fitbenchmarking/utils/timer.py). If you can report issues you're having with max runtime on this issue #1518, we can look to fix in a separate PR